components/service/glean/src/test/java/mozilla/components/service/glean/GleanTest.kt

```diff
@@ -0,0 +1,163 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.glean
+
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotNull
+import org.junit.Assert.assertNull
+import org.junit.Test
+
+import org.json.JSONObject
+
+import org.mockito.ArgumentMatchers.anyString
+import org.mockito.Mockito.doAnswer
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.times
+
+import mozilla.components.service.glean.net.HttpPingUploader
+import mozilla.components.service.glean.storages.EventsStorageEngine
+import mozilla.components.service.glean.storages.StringsStorageEngine
+
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+class GleanTest {
+    @Test
+    fun `disabling metrics should record nothing`() {
+        val stringMetric = StringMetricType(
+                disabled = false,
+                category = "telemetry",
+                lifetime = Lifetime.Application,
+                name = "string_metric",
+                sendInPings = listOf("store1")
+        )
+        Glean.setMetricsEnabled(false)
+        try {
```

Comment:
Because we need to reset the global state that changes here (`setMetricsEnabled`).  Otherwise other unrelated tests start failing randomly (depending on order of test run).

Reply:
Nothing is throwing on purpose.  What I'm indicating is that no matter what happens in this test function, the global state needs to be reset at the end (in the `finally` block).  The behavior that this prevents is a failure in this test leading to a failure in other tests simply because they come after this one.

If there is a way to set a `@Before` on every test in the test suite, that would be a viable alternative.  But I think `@Before` just affects tests in the same fixture or file.

Reply:
It does like like, however, that we could use `@After`, which might be cleaner / more obvious than `try/finally`.  I'll play with that.